kani-compiler/src/codegen_cprover_gotoc/compiler_interface.rs

```diff
@@ -175,14 +176,32 @@ impl CodegenBackend for GotocCodegenBackend {
             let outputs = tcx.output_filenames(());
             let base_filename = outputs.output_path(OutputType::Object);
             let pretty = self.queries.lock().unwrap().get_output_pretty_json();
+
+            let write_goto_binary = gcx.queries.get_write_goto_binary();
+            if write_goto_binary {
+                // write a GOTO binary ourselves
+                write_goto_binary_file(
+                    &gcx.symbol_table,
+                    &base_filename.with_extension(ArtifactType::SymTabGoto),
+                )
+                .unwrap(); // is this the right way to handle errors here ?
+            } else {
+                write_file(&base_filename, ArtifactType::SymTab, &gcx.symbol_table, pretty);
+            }
+
+            write_file(&base_filename, ArtifactType::SymTab, &gcx.symbol_table, pretty);
```

Comment:
Is there a reason why you split this code? I would suggest that we keep this at the end, together with the code that invokes the symtab2gb.

Reply:
fixed